crawl-ref/source/spl-book.cc

```diff
@@ -874,6 +870,11 @@ static bool _learn_spell_checks(spell_type specspell, bool wizard = false)
 */
 bool learn_spell(spell_type specspell, bool wizard)
 {
+    if (god_loathes_spell(specspell, you.religion))
+        mprf(MSGCH_WARN,"Casting this spell will cause instant excommunication!");
+    else if (god_hates_spell(specspell, you.religion))
```

Comment:
Good ideas.  

I added docstrings for the two functions, and one for god_hates_spellcasting.

I took a look at spl-cast.cc, but didn't see anything that jumped out at me, but I'm not sure my eyes are trained enough yet to catch the things I should be looking for.

I liked your suggestion on the warnings.  They've been made a bit more generic/vague, but I think it still gets the point across